### ISSUE TITLE
Need to speed up slow tests

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#3628]
We have two sequential tests:
```
        599 - rising_melt_blob_freezing (Timeout)                                                                                           
        667 - statistics_output (Timeout)   
```
that seem to take a very, very long time since we switched to deal.II 9.2 (``statistics_output`` takes 45min+ for me).

I am not sure what is going on here...

**Comment by tjhei** [comment]
rising_melt_blob_freezing:
```
| Total wallclock time elapsed since start     |       815s |            |
|                                              |            |            |
| Section                          | no. calls |  wall time | % of total |
+----------------------------------+-----------+------------+------------+
| Assemble Stokes system           |        96 |       137s |        17% |
| Assemble composition system      |       192 |       265s |        33% |
| Assemble temperature system      |        96 |       139s |        17% |
| Build Stokes preconditioner      |        96 |      79.3s |       9.7% |

```

**Comment by tjhei** [comment]
this should be fixed with #3618

**Comment by tjhei** [comment]
new times with #3618:
```
1/1 Test #599: rising_melt_blob_freezing ........   Passed  229.85 sec
1/1 Test #667: statistics_output ................   Passed  575.50 sec
```
So they are much better but still likely responsible in the order of 8% of the total test time (assuming 13 minutes of real time with 12 cores).

**Comment by tjhei** [comment]
new statistics_output timing:
```
+----------------------------------------------+------------+------------+
| Total wallclock time elapsed since start     |       570s |            |
|                                              |            |            |
| Section                          | no. calls |  wall time | % of total |
+----------------------------------+-----------+------------+------------+
| Assemble Stokes system           |        16 |      75.9s |        13% |
| Assemble composition system      |        32 |       190s |        33% |
| Assemble temperature system      |        16 |      99.9s |        18% |
| Build Stokes preconditioner      |        16 |      49.5s |       8.7% |
```

**Comment by tjhei** [comment]
Other slow tests (prebuild timed out probably due to parallel load), see https://jenkins.geodynamics.org/blue/organizations/jenkins/aspect/detail/PR-3631/1/pipeline
Run sequentially in phase 2 they take:
```
181/802 Test #181: continental_extension ..............................................   Passed  345.64 sec
603/802 Test #603: segregation_heating ................................................   Passed  291.77 sec
667/802 Test #667: statistics_output ..................................................   Passed  536.98 sec
795/802 Test #795: vof_mpi_amr_no_dof .................................................   Passed  463.06 sec
```
Together, they add up to slowing down a test run by 27 minutes.

**Comment by tjhei** [comment]
closed by #3639
